google/cloud/security/common/util/email_util.py

```diff
@@ -116,7 +103,7 @@ def send(self, email_sender=None, email_recipient=None,
         if not email_sender or not email_recipient:
             LOGGER.warn('Unable to send email: sender=%s, recipient=%s',
                         email_sender, email_recipient)
-            raise util_errors.EmailSendError
+            return
```

Comment:
This is handled upstream to capture a bit more detail of what is not able to be sent.  If you want to keep the logging here, can you please add the detail here, perhaps logging the email subject?

https://github.com/GoogleCloudPlatform/forseti-security/blob/dev/google/cloud/security/notifier/pipelines/email_inventory_snapshot_summary_pipeline.py#L92

If the EmailSendError will not be used / raised, can you please clean remove the exception and from upstream?

Reply:
Hm, interesting -- I think I'll tweak the code so that we keep the EmailSendError and handle it upstream for the Scanner summary. That's where the stacktrace is showing up.

Reply:
Tweaked the scanner and violation email notification pipelines so that they catch the EmailSendError.